connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -750,7 +751,8 @@ class SparkConnectPlanner(val session: SparkSession) {
         }
         cols
       }
-      Deduplicate(groupCols, queryExecution.analyzed)
+      if (rel.getWithinWatermark) DeduplicateWithinWatermark(groupCols, queryExecution.analyzed)
```

Comment:
Actually, should we have a dedicated protobuf message for `DeduplicateWithinWatermark`? Seems like we don't also share the same type of logical plan. Do you have any preference on this, @grundprinzip and @amaliujia?

Reply:
Okay, just noticed that it was discussed at https://github.com/apache/spark/pull/40834#discussion_r1170693559. I will go ahead and merge if you guys don't have preference. I don't feel strongly about this either.

Reply:
@HyukjinKwon thanks. This is much simpler code wise. 1:1 for logical plans is not strictly required, I hope.

Reply:
If there is anything wrong, I think deprecating a field is easier than deprecating a new relation type. Probably starting from this by adding a new flag way is a good beginning.